ppx/browser/ppx_deriving_json_js.ml

```diff
@@ -65,13 +65,13 @@ module Of_json = struct
   let ensure_json_object ~loc x =
     [%expr
       if Stdlib.not [%e eis_json_object ~loc x] then
-        Ppx_deriving_json_runtime.of_json_error
+        Ppx_deriving_json_runtime.of_json_msg_error
           [%e estring ~loc (sprintf "expected a JSON object")]]
 
   let ensure_json_array_len ~loc n len =
     [%expr
       if Stdlib.( <> ) [%e len] [%e eint ~loc n] then
-        Ppx_deriving_json_runtime.of_json_error
+        Ppx_deriving_json_runtime.of_json_msg_error ~json:x
```

Comment:
```suggestion
        Ppx_deriving_json_runtime.of_json_msg_error ~json:[%e x]
```
let's pass `x` to this function so we can use like this ^

Reply:
Its not that simple. I tried and I could not do it. The reason is that this function is called from `derive_of_variant_case` which does not have `x`, and `derive_of_variant_case` is directly given to `deriving_of` which uses objects and that I did not really understand. If you can find a way to make this cleaner it would be great.

Reply:
I was referring to the immediate call site 
```
[%e ensure_json_array_len ~loc 2 [%expr len] [%expr x]];
```
so that the assumptions about what names are defined do not spread to helper functions

Reply:
okay why not.

Reply:
This is done